storage/mysql/log_storage.go

```diff
@@ -265,6 +266,12 @@ func (t *logTX) QueueLeaves(leaves []trillian.LogLeaf) error {
 		if len(leaf.LeafHash) != t.ts.hashSizeBytes {
 			return fmt.Errorf("Queued leaf must have a hash of length %d", t.ts.hashSizeBytes)
 		}
+
+		// Validate the hash as a consistency check that the data was received OK. Note: at
+		// this stage it is not a merkle tree hash for the leaf
+		if !bytes.Equal(trillian.NewSHA256().Digest(leaf.LeafValue), leaf.LeafHash) {
```

Comment:
Do the `got, want` thing here to make the log message (infinitesimally) more helpful?